src/plugins/intel_npu/src/backend/src/zero_executor.cpp

```diff
@@ -120,6 +126,35 @@ ZeroExecutor::ZeroExecutor(const std::shared_ptr<const ZeroInitStructsHolder>& i
     _logger.debug("ZeroExecutor::ZeroExecutor - performing hostSynchronize");
     fence.hostSynchronize();
     _logger.debug("ZeroExecutor::ZeroExecutor - hostSynchronize completed");
+
+    if (config.has<WORKLOAD_TYPE>()) {
+        setWorkloadType(config.get<WORKLOAD_TYPE>());
+    }
+}
+
+void ZeroExecutor::updateConfig(const Config& config) {
```

Comment:
If there's no point in storing the config, maybe we don't need this method either. We may leverage `setWorkloadType` instead.

Reply:
I agree, the config in zero executor is used only in the constructor. I don't know the reasons why it was stored previously. I reverted the changes done to config and compiled_model will call setWorkloadType directly. IMO config can be removed entirely in a future PR.